rs/cli/src/js.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::Result;
+use std::process::Command;
+
+const SAILS_JS_CLI_VERSION: &str = "0.1.0";
+
+pub struct JsClientGenerator {
+    idl_path: String,
+    out_path: String,
+    program_name: String,
+}
+
+impl JsClientGenerator {
+    pub fn new(idl_path: String, out_path: String, program_name: String) -> Self {
+        Self {
+            idl_path,
+            out_path,
+            program_name,
+        }
+    }
+
+    pub fn generate(&self) -> Result<()> {
+        check_node()?;
+        check_npx()?;
+
+        let out_path = self.out_path.clone();
+        let program_name = self.program_name.clone();
+        let idl_path = self.idl_path.clone();
+
+        let mut child = Command::new("npx")
+            .arg(format!("sails-js-cli@{}", SAILS_JS_CLI_VERSION))
+            .arg("generate")
+            .arg(idl_path)
+            .arg("-o")
+            .arg(out_path)
+            .arg("-n")
+            .arg(program_name)
+            .spawn()
+            .expect("Failed to run npx sails-js-cli");
+
+        let status = child.wait().expect("An error occured");
+
+        if !status.success() {
+            panic!("Failed to generate JS client");
```

Comment:
Please instead of using `panic`, `expect`, etc., use Result to return the error s it can printed out by the program in standard way. This comment is applicable to the entire mod